src/uthash.h

```diff
@@ -177,24 +193,45 @@ do {
 
 #define HASH_MAKE_TABLE(hh,head)                                                 \
 do {                                                                             \
+  int _mem_ok = 1;                                                               \
   (head)->hh.tbl = (UT_hash_table*)uthash_malloc(sizeof(UT_hash_table));         \
-  if (!(head)->hh.tbl) {                                                         \
+  if (!((head)->hh.tbl)) {                                                       \
+    if (HASH_NOMEM_OK) {                                                         \
+      break;                                                                     \
+    }                                                                            \
     uthash_fatal("out of memory");                                               \
```

Comment:
You are saying it's an anathema to check `errno`/return values from every library call, but wait, wouldn't very bad things happen if you didn't? Usually in a chain of functions if one fails, all others wouldn't make sense, and it takes craft to either make the rest be noops, or unwind at the end. Yes, it's very tedious and probably are responsible for a few folks to change their professions. That's why exceptions are so popular, because they let you not check the result, and just deal with any error in a concentrated place.

But, there in the above I convinced myself. People can sure want to use exceptions there if they are using it in C++ code. Even breaking out of their do{}while() loops that so conveniently mimic the exception model (labels required).

How about this : if when `HASH_NOMEM_OK` is defined, `uthash.h` also defines, if not set, `uthash_failed`, with the arguments of `element` (to be filled in with the element that is somehow involved in the failure), to do nothing (the caller has to define it with a meaningful filling). Then, at the end of any potential clean up, and after `mem_failed` is set to 1, the `uthash_failed` is called.